### PR TITLE
Multiple additional relative date filters

### PR DESCRIPTION
**Pull request by jamiemonksuk**
I've added these for a client and I think it's the finishing touch to this filtering aspect that I mentioned I was still unhappy with when we started amending it.

### CODE REVIEW

File: src/sprout/Controllers/Admin/ManagedAdminController.php [hunk 1]
```diff
@@ -856,9 +866,40 @@ protected function _getRefineClause($key, $val, array &$query_params)
...
-                    $end = date('Y-m-d 23:59:59', $yesterday);
-                    return "item.{$key} BETWEEN '{$start}' AND '{$end}'";
+                    $query_params[] = date('Y-m-d 00:00:00', $yesterday);
+                    $query_params[] = date('Y-m-d 23:59:59', $yesterday);
+                    return "item.{$key} BETWEEN ? AND ?";
+
+                } else if ($val == 'THIS_WEEK') {
+                    $query_params[] = date('Y-m-d 00:00:00', strtotime('last monday'));
+                    $query_params[] = date('Y-m-d 23:59:59', strtotime('next sunday'));
+                    return "item.{$key} BETWEEN ? AND ?";
```
Comment (gwillz) [inline src/sprout/Controllers/Admin/ManagedAdminController.php:876]: I assume we don't have any locale settings for a 'week start day' anywhere?

File: src/sprout/Helpers/Constants.php [hunk 2]
```diff
@@ -96,11 +96,17 @@ public static function reverseLookup($class, $prefix, $value)
...
-        '1 WEEK' => 'This week',
-        '1 MONTH' => 'This month',
-        '3 MONTH' => 'This quarter',
-        '1 YEAR' => 'This year',
+        'THIS_WEEK' => 'This week',
+        'THIS_MONTH' => 'This month',
+        'THIS_YEAR' => 'This year',
+        'WEEK_TO_DATE' => 'Week to date',
+        'MONTH_TO_DATE' => 'Month to date',
+        'YEAR_TO_DATE' => 'Year to date',
```
Comment (gwillz) [inline src/sprout/Helpers/Constants.php:109]: I do fear overcrowding this but I wonder there's room for a set of LAST_WEEK/MONTH/YEAR ?

Or perhaps there's a way we can make this pluggable so we can customise these ranges per site. That's probably a better idea.

**Review by gwillz — APPROVED** [review]
I like em. They're better than what we had.

We can riff on some kind of event/config thing later.

File: src/sprout/Helpers/Constants.php (hunk 2 above)
Comment (jamiemonksuk) [reply src/sprout/Helpers/Constants.php:109]: I thought about that, but I couldn't think of anything off the top of my head and was a bit time poor to be hunting for it..

Happy to add it in if @aitken85 or anyone else can think of something

**Pull request merged**